pkg/worker/worker.go

```diff
@@ -420,7 +441,15 @@ func (w *Worker) printBuildInfo() {
 //Run runs the worker and returns error if any.
 func (w *Worker) Run() (bool, error) {
 	var success bool
-	if err := w.cleanupOldBuilds(); err != nil {
+	err := w.checkForNewerBuilds()
+	if err != nil {
+		return false, err
+	}
+	err = w.sendCancelMessage()
+	if err != nil {
+		return false, fmt.Errorf("failed to send cancel message %w", err)
+	}
+	if err = w.cleanupOldBuilds(); err != nil {
```

Comment:
This is to check for new build and stop the ongoing build?

Reply:
yes